### PR TITLE
Retry appveyor download

### LINKED ISSUE #577: AppVeyor: make the downloading more robust
I reported the issue here:

http://help.appveyor.com/discussions/problems/2753-rawgithubusercontentcom-gives-server-unavailable-sporadically

The conclusion is that we need to try several times, with a timeout of 2s, 4s, 8s, ... up to a minute or so.

### PR DESCRIPTION
**Pull request by isuruf**
Based on https://gist.github.com/FeodorFitsner/508b71250295590e6408

Fixes #577

### CODE REVIEW

**Comment by certik** [comment]
I would also add a progressive time out, i.e. wait 1s, then 2s, 4s, 8s, 16s. I would do max retries 6x. That way it takes 31s total in the worst case, which is not bad, while giving github a chance to succeed at the same time.

**Comment by Sumith1896** [comment]
Looks good

File: bin/appveyor-download.cmd [hunk 1]
```diff
@@ -0,0 +1,18 @@
+@echo off
+rem initiate the retry number
+set retryNumber=0
+set maxRetries=6
+set time=1
+
+:DOWNLOAD
+timeout %time% > NUL
```
Comment (certik) [inline bin/appveyor-download.cmd]: Now it waits 1s for the first time, doesn't it?

Comment (isuruf) [inline bin/appveyor-download.cmd]: Fixed

**Comment by certik** [comment]
So the sequence of timeouts is 1, 3, 7, 15, 31. That's fine.

+1 to merge if tests pass.

**Comment by isuruf** [comment]
I rearranged the lines, so that an error is detected

**Comment by isuruf** [comment]
This is ready now.

**Comment by certik** [comment]
Thanks!

**Pull request merged**